java/vertx-proton/src/main/java/net/ssorj/quiver/QuiverVertxProton.java

```diff
@@ -0,0 +1,177 @@
+        int credit = Integer.parseInt(args[7]);
+
+        if (!CLIENT.equalsIgnoreCase(mode)) {
+            throw new RuntimeException("This impl currently supports client mode only");
+        }
+
+        final boolean sender;
+        if (SEND.equalsIgnoreCase(operation)) {
+          sender = true;
+        } else if (RECEIVE.equalsIgnoreCase(operation)) {
+          sender = false;
+        } else {
+          throw new java.lang.IllegalStateException("Unknown operation: " + mode);
+        }
+
+        CountDownLatch completionLatch = new CountDownLatch(1);
+        URI uri = new URI("amqp://" + domain);
+
+        String hostname = uri.getHost();
+        int port = uri.getPort();
+        if(port == -1) {
+          port = DEFAULT_AMQP_PORT;
+        }
+
+        Vertx vertx = Vertx.vertx();
+
+        ProtonClient client = ProtonClient.create(vertx);
+        client.connect(hostname, port, res -> {
+          if (res.succeeded()) {
+            ProtonConnection connection = res.result();
+
+            if (sender) {
+              send(connection, path, messages, bytes, completionLatch);
+            } else {
+              receive(connection, path, messages, credit, completionLatch);
+            }
+          } else {
+            res.cause().printStackTrace();
+            completionLatch.countDown();
+          }
+        });
+
+        // Await the operations completing, then shut down the Vertx instance.
+        completionLatch.await();
+
+        vertx.close();
+    }
+
+    private static void send(ProtonConnection connection, String address, int messages, int bytes, CountDownLatch latch) {
+      connection.open();
+
+      byte[] payloadContent = new byte[bytes];
+      Arrays.fill(payloadContent, (byte) 120);
+
+      AtomicLong count = new AtomicLong(1);
+
+      ProtonSender sender = connection.createSender(address);
+      sender.sendQueueDrainHandler(s -> {
+        while (!sender.sendQueueFull() && count.get() <= messages) {
+          Message msg = Message.Factory.create();
+
+          msg.setMessageId(UnsignedLong.valueOf(count.get()));
+
+          msg.setBody(new Data(new Binary(payloadContent)));
+
+          Map<String, Object> props = new HashMap<>();
+          msg.setApplicationProperties(new ApplicationProperties(props));
+          long stime = System.currentTimeMillis();
+          props.put("SendTime", stime);
+
+          sender.send(msg);
+          if(count.getAndIncrement() >= messages) {
+            connection.closeHandler(x -> {
+              latch.countDown();
+            });
+            connection.close();
+          };
+        }
+      });
+      sender.open();
+    }
+
+    private static void receive(ProtonConnection connection, String address, int messages, int credits, CountDownLatch latch) {
+      connection.open();
+
+      //TODO: adjust? The writer is [needlessly] synchronizing every write, the buffer may flush more/less often than desired?.
```

Comment:
Okay, right.  I'm expecting the buffered writer to flush itself.  Perhaps we should also select an explicit buffer size.